solidity/dashboard/src/components/Timeline.jsx

```diff
@@ -0,0 +1,51 @@
+import React, { useContext } from "react"
+
+const TimelineContext = React.createContext({
+  // May be useful in the future, eg. if we would like to be able to display the
+  // timeline events on alternating sides.
+  position: "left",
+})
+
+const useTimelineContext = () => {
+  const context = useContext(TimelineContext)
+  if (!context) {
+    throw new Error("TimelineContext used outside of Timeline component")
+  }
+  return context
+}
+
+const Timeline = ({ children, position = "left" }) => {
+  return (
+    <TimelineContext.Provider value={{ position: position }}>
+      <div className="timeline">{children}</div>
+    </TimelineContext.Provider>
+  )
+}
+const TimelineElement = ({ position, children, className = "" }) => {
+  const { position: timelinePosition } = useTimelineContext()
+  const _position = position || timelinePosition
+  return (
+    <div className={`timeline__element timeline__element--${_position} ${className}`}>
+      {children}
+    </div>
+  )
+}
+Timeline.Element = TimelineElement
+
+Timeline.ElementBreakpoint = ({
+  children,
+  className = "",
+  // start, center
```

Comment:
According to the Figma design system currently, we need `start` and `center`, so I think we can skip this for now. Thoughts?

Reply:
I think this comment is outdated now because the dots are always on the start position (according to the Figma views). We can add support for this position when needed.